src/components/modals/LoginModal.tsx

```diff
@@ -39,23 +40,31 @@ export default function LoginModal({
 
   const isLoading = loadingRequestToken || hasStartCaptcha
 
-  const onSubmit = async (e: SyntheticEvent) => {
-    e.preventDefault()
-    beforeLogin?.()
-    if (await login(privateKey)) {
-      afterLogin?.()
-      setPrivateKey('')
-      props.closeModal()
-    } else {
-      toast.custom((t) => (
-        <Toast
-          t={t}
-          title='Login Failed'
-          description='The grill secret key you provided is not valid'
-        />
-      ))
+  const onSubmit =
+    (authMethod: AUTHENTICATION_METHODS) => async (e: SyntheticEvent) => {
+      e.preventDefault()
+      beforeLogin?.()
+      let loginRes = false
+      if (authMethod === 'Anonymous') {
+        await loginAnonymously(privateKey).then(() => (loginRes = true))
+      } else {
+        await login().then(() => (loginRes = true))
```

Comment:
for this one, I think its better to have both login methods return true/false
and for this line, you just
```
loginRes = await login()
```

also, using await and `then` in the same part is also redundant
so if you want to use this way, its better to have
```
await login()
loginRes = true
```

but anyway, I think its better to have the first approach, where both login methods return boolean as indicator success or not

Reply:
Good one, in some cases the address retuned by the login is used to request captcha

<img width="528" alt="Screenshot 2023-05-17 at 11 49 47" src="https://github.com/dappforce/grillchat/assets/56865496/9e7cef41-b6f2-4eb1-812b-4e2fce22a618">

I had to make loginRes: string | boolean... should still serve similar purpose